ssl-config/src/main/java/com/palantir/remoting/ssl/SslConfiguration.java

```diff
@@ -28,18 +30,44 @@
 @Value.Style(visibility = ImplementationVisibility.PACKAGE)
 public abstract class SslConfiguration {
 
-    @Value.Parameter
-    public abstract TrustStoreConfiguration trust();
+    public abstract URI trustStorePath();
 
-    @Value.Parameter
-    public abstract Optional<KeyStoreConfiguration> key();
+    @SuppressWarnings("checkstyle:designforextension")
+    @Value.Default
+    public String trustStoreType() {
+        return "JKS";
+    }
+
+    public abstract Optional<URI> keyStorePath();
+
+    public abstract Optional<String> keyStorePassword();
+
+    @SuppressWarnings("checkstyle:designforextension")
+    @Value.Default
+    public String keyStoreType() {
+        return "JKS";
+    }
+
+    // alias of the certificate to use in the key store.
+    // If absent, first entry returned by key store is used.
+    public abstract Optional<String> certAlias();
```

Comment:
Even though this is what Dropwizard uses I think it's ambguous here and would prefer something appropriately specific, like keyStoreKeyAlias